### ISSUE TITLE
aria-hidden with focusable content: Overlap with several rules

### ISSUE THREAD

**Issue by kasperisager** [auto-wcag/auto-wcag#301]
Per https://github.com/auto-wcag/auto-wcag/issues/281 and https://github.com/auto-wcag/auto-wcag/pull/269#issuecomment-428519520, `SC4-1-2-aria-hidden-focus` currently overlaps with several rules:

- `SC3-3-2-form-field-has-name`
- `SC2-4-4-link-has-name`
- `SC4-1-2-button-has-name`

It is possible for an element to be applicable to and pass any of the above rules, and then fail `SC4-1-2-aria-hidden-focus` for the same reason the element was applicable to the other rules in the first place. Together, these rules are effectively saying that, sure, you must have accessible names for these types of elements if they're focusable, but the accessible names won't ever be announced to AT because, guess what, you haven't included the elements in the accessibility tree. That seems a little backwards.

I'm thinking it would make more sense for `SC4-1-2-aria-hidden-focus` to first flag elements that are focusable, but not included in the accessibility tree, and if authors then do decide to include the elements in the accessibility tree, the other rules should then kick in to ensure that the elements also have accessible names. 

I have similar concerns for these rules, which also apply to focusable elements that aren't included in the accessibility tree:

- `SC1-2-2-audio-captions`
- `SC1-3-5-autocomplete-valid`

**Comment by annethyme** [comment]
Handled by #310. Please review.

**Comment by annethyme** [comment]
Handled by #264. Please review.
